src/components/elementView/detailedView/assetView/assetView.tsx

```diff
@@ -202,73 +222,80 @@ export class AssetView extends React.PureComponent<
+          <DecommissionedPropertiesView data={this.state.asset} />
+        ) : null}
         <PropertiesView data={this.state.asset} />
         <div className="propsview">
           <h3>Network Connections</h3>
 
           {this.state.asset.model &&
-          this.state.asset.model.network_ports &&
-          this.state.asset.model.network_ports.length !== 0 ? (
-            <div className="network-connections">
-              <table className="bp3-html-table bp3-html-table-bordered bp3-html-table-striped">
-                <tr>
-                  <th>Network Port</th>
-                  <th>Mac Address</th>
-                  <th>Destination Asset</th>
-                  <th>Destination Port</th>
-                </tr>
-                <tbody>
-                  {this.state.asset.model.network_ports.map((port: string) => {
-                    var connection = this.getNetworkConnectionForPort(port);
-                    return (
-                      <tr>
-                        {" "}
-                        <td>{port}</td>
-                        <td>{this.state.asset.mac_addresses[port]}</td>{" "}
-                        {connection
-                          ? [
+            this.state.asset.model.network_ports &&
+            this.state.asset.model.network_ports.length !== 0 ? (
+              <div className="network-connections">
+                <table className="bp3-html-table bp3-html-table-bordered bp3-html-table-striped">
+                  <tr>
+                    <th>Network Port</th>
+                    <th>Mac Address</th>
+                    <th>Destination Asset</th>
+                    <th>Destination Port</th>
+                  </tr>
+                  <tbody>
+                    {this.state.asset.model.network_ports.map((port: string) => {
+                      var connection = this.getNetworkConnectionForPort(port);
+                      return (
+                        <tr>
+                          {" "}
+                          <td>{port}</td>
+                          <td>{this.state.asset.mac_addresses ? this.state.asset.mac_addresses[port] : null}</td>{" "}
+                          {connection
+                            ? [
                               <td
-                                className="asset-link"
-                                onClick={(e: any) => {
-                                  const id = this.getAssetIdFromHostname(
-                                    connection!.destination_hostname!
-                                  );
-                                  if (id) {
-                                    this.redirectToAsset(id);
-                                  }
-                                }}
+                                className={this.state.asset.decommissioning_user ? undefined
+                                  : "asset-link"}
+                                onClick={this.state.asset.decommissioning_user ? undefined
```

Comment:
quick Q regarding onClick: does undefined actually work or does the program crash if you try to click? Normally if we don't want a button to do something on click, Claire and I have just been filling in an empty arrow function () => { }

Reply:
Undefined works as far as I can tell. I think I tried null and it crashed but undefined seemed to work